File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    result=[0, 0]
    
    while True:
        result[1]+=(len(s)-eliminateZero(s))
        s=changeToBinary(eliminateZero(s))
        result[0]+=1
        
        if s=="1":
            return result
    

def eliminateZero(x):
    x=str(x)
    numOfOne=0
    for i in range(len(x)):
        if x[i]=="1":
            numOfOne+=1
    return numOfOne

def changeToBinary(num):
    arr=[]
    while num>0:
        arr.append(num%2)
        num=num//2
    arr.reverse()
    return ''.join(map(str, arr))
    

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    let result=[0, 0];
    while (s!=="1") {
        result[1]+=s.length-eliminateZero(s);
        s=changeToBinary(eliminateZero(s));
        result[0]+=1;
    }
    return result;
}


function eliminateZero(x) {
    let count=0;
    for (let i=0; i<x.length; i++) {
        if (x[i]=="1")
            count+=1;
    }
    return count;
}

function changeToBinary(num) {
    let arr=[];
    while (num>0) {
        arr.push(num%2);
        num=Math.floor(num/2);
    }
    return arr.reverse().join('');
}